File: src/main/kotlin/demo/kotliln/boardgame/service/rating/RateEntity.java
package demo.kotliln.boardgame.service.rating;
import lombok.*;

import javax.persistence.*;

import static javax.persistence.GenerationType.IDENTITY;

@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PACKAGE)
@Entity
@Getter
public class RateEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    @Column(name = "boardGameName", nullable = false)
    private String boardGameName;


    @Column(name = "rate", nullable = false)
    private Double rate;

}

File: src/main/kotlin/demo/kotliln/boardgame/service/rating/entities/RateEntity.kt
package demo.kotliln.boardgame.service.rating.entities

import javax.persistence.*

@Entity
data class RateEntity (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long?,

    @Column(name = "boardGameName", nullable = false)
    val boardGameName: String,

    @Column(name = "rate", nullable = false)
    val rate: Double

)